Update publishing-java-packages-with-gradle.md

mahozad opened the pull request:
### Why:

Closes #8048

### What's being changed:

- Merge a somewhat duplicate sentence about setting environment variables in the deploy step
- Fix an error in a repository name (Maven Central instead of GitHub Packages) (may have happened because of a copy-paste)
- Fix another paragraph that only referred to configuration of GitHub Packages while it was about configuring both Maven Central and GitHub Packages (may have happened because of a copy-paste)

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

mahozad committed:
Update publishing-java-packages-with-gradle.md

- Merge a somewhat duplicate sentence about setting environment variables in the deploy step
- Fix an error in a repository name (Maven Central instead of GitHub Packages) (may have happened because of a copy-paste)
- Fix another paragraph that only referred to configuration of GitHub Packages while it was about configuring both Maven Central and GitHub Packages (may have happened because of a copy-paste)

ramyaparimi commented:
@mahozad Thanks so much for opening a PR! I'll get this triaged for review :zap:

mahozad committed:
Fix another incorrect name in the code

The code publishes to both Maven Central AND GitHub Packages so its name would better be "Publish to Maven Central and GitHub Packages" or like previous code snippets simply "Publish package"

Caglar1881 reviewed (changes requested):
👍

Caglar1881 reviewed (changes requested):
😊

soffieprincess09 commented:
> ### Why:
> Closes #8048
> 
> ### What's being changed:
> * Merge a somewhat duplicate sentence about setting environment variables in the deploy step
> * Fix an error in a repository name (Maven Central instead of GitHub Packages) (may have happened because of a copy-paste)
> * Fix another paragraph that only referred to configuration of GitHub Packages while it was about configuring both Maven Central and GitHub Packages (may have happened because of a copy-paste)
> 
> ### Check off the following:
> * [ ]  I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
> * [ ]  For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).
> 
> ### Writer impact (This section is for GitHub staff members only):
> * [ ]  This pull request impacts the contribution experience
>   
>   * [ ]  I have added the 'writer impact' label
>   * [ ]  I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

soffieprincess09 reviewed (dismissed):
``

lecoursen commented on content/actions/guides/publishing-java-packages-with-gradle.md:
```diff
@@ -75,9 +75,7 @@ publishing {
-With this configuration, you can create a workflow that publishes your package to the Maven Central Repository by running the `gradle publish` command. You’ll also need to provide environment variables that contain the username and password to authenticate to the repository.
-
-In the deploy step, you’ll need to set environment variables for the username and password or token that you use to authenticate to the Maven repository. For more information, see "[Creating and using encrypted secrets](/github/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets)."
+With this configuration, you can create a workflow that publishes your package to the Maven Central Repository by running the `gradle publish` command. In the deploy step, you’ll also need to set environment variables for the username and password or token that you use to authenticate to the Maven repository. For more information, see "[Creating and using encrypted secrets](/github/automating-you ...
```
```suggestion
With this configuration, you can create a workflow that publishes your package to the Maven Central Repository by running the `gradle publish` command. In the deploy step, you’ll need to set environment variables for the username and password or token that you use to authenticate to the Maven repository. For more information, see "[Creating and using encrypted secrets](/github/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets)."
```

lecoursen reviewed (dismissed):
This looks great, thanks for catching thee @mahozad! I'm going to apply one tiny change and then merge. ⚡

lecoursen committed:
Add :nail_care:

lecoursen commented on content/actions/guides/publishing-java-packages-with-gradle.md:
```diff
@@ -144,7 +142,7 @@ publishing {
 {% endraw %}
 
-With this configuration, you can create a workflow that publishes your package to the Maven Central Repository by running the `gradle publish` command.
+With this configuration, you can create a workflow that publishes your package to GitHub Packages by running the `gradle publish` command.
```
```suggestion
With this configuration, you can create a workflow that publishes your package to {% data variables.product.prodname_registry %} by running the `gradle publish` command.
```

haaami01 replied:
Ezzel a konfigurációval a `gradle publish` parancs futtatásával létrehozhat egy munkafolyamatot, amely a Maven Central Repository-ban és a {% data variables.product.prodname_registry %}-ban is közzéteszi a csomagot

lecoursen committed:
Use product name variable

lecoursen committed:
Merge branch 'main' into patch-2
